tutorials/theory_buoyancy.md

```diff
@@ -0,0 +1,98 @@
+to match your expected center of buoyancy.
+
+The buoyancy force is proportional to the volume of air in the vehicle according
+to this equation:
+
+$$volume\\_neutral = \frac{mass}{waterDensity}$$
+
+Pure water's density is `1000 kgm^-3` and seawater's density is `1025 kgm^-3`
+approximately. The mass of your vehicle is something that you'll decide or know
+by adding the mass of all your links. With that in mind, you'll be able to
+compute the volume of air that will make your vehicle neutrally buoyant.
+
+If your vehicle is neutrally buoyant, `volume_neutral` should be the total
+amount of volume that you should have by adjusting the size of your
+`<collision>` elements. If your vehicle is not neutrally buoyant, this value
+will serve as a reference. If the total volume is smaller than `volume_neutral`,
+the vehicle will sink. Otherwise, it will move up.
+
+The `<GAZEBO_INSTALL_PATH>/examples/worlds/buoyancy.sdf` SDF file (included with
+Gazebo) contains an example with three submarines. The first submarine is
+neutrally buoyant, the second sinks, and the third floats. Run the following
+command to see an example of a buoyancy plugin configured with a uniform fluid
+density:
+
+```bash
+gz sim buoyancy.sdf
+```
+
+# Graded buoyancy
+
+Often when simulating a maritime environment one may need to simulate both
+surface and underwater vessels. This means the buoyancy plugin needs to
+take into account two different fluids. One being water with a density of
+`~1000 kgm^-3` and another being air with a very light density of say `1 kgm^-3`.
+An example for such a configuration may be found in the
+`<GAZEBO_INSTALL_PATH>/examples/worlds/graded_buoyancy.sdf` world (included
+with Gazebo).
+
+```bash
+gz sim graded_buoyancy.sdf
+```
+
+You should be able to see a sphere bobbing up and down undergoing simple
+harmonic motion on the surface of the fluid (this is expected behaviour
+as the harmonic motion is usually damped by the hydrodynamic forces. See the
+hydrodynamics tutorial for an example of how to use it). The key part of this is
+
+```xml
+<graded_buoyancy>
+  <default_density>1000</default_density>
+  <density_change>
+    <above_depth>0</above_depth>
+    <density>1</density>
+  </density_change>
+</graded_buoyancy>
+```
+
+The default density tag says that by default the world has a fluid density
+of `1000 kgm^-3`. This essentially states that by default the world is filled
+with dihydrogen monoxide (aka water). The `<density_change>` tag essentially
+establishes the fact that there is a another fluid. The `<above_depth>` tag says
+that above z=0 there is another fluid with a different density. The density of
+that fluid is defined by the `<density>` tag. We will be simulating air with a
+fluid density of `1 kgm^-3`.
+
+# Known limitations
+
+When the buoyancy plugin is configured in `graded buoyancy mode`, only `<box>`
+and `<sphere>` collision geometries are supported.
```

Comment:
Also add a note that it relies on collision meshes. This can be annoying for end users who want to accurately model collision and buoyancy but the two have two different types of behaviour.

Reply:
Is your suggestion captured in this paragraphs?

https://github.com/gazebosim/gz-sim/pull/2258/files#diff-378e68a808fda7960b03c7a17b40fd5945fb274e584ed96850c3dac10d6bd4cbR16-R30